python/whylogs/datasets/weather.py

```diff
@@ -0,0 +1,229 @@
+import os
+from dataclasses import dataclass
+from datetime import date, datetime, timedelta
+from importlib import resources
+from logging import getLogger
+from typing import Iterable, Optional, Tuple, Union
+
+import pandas as pd
+
+from whylogs.datasets.base import Batch, Dataset
+from whylogs.datasets.configs import BaseConfig, DatasetConfig, WeatherConfig
+from whylogs.datasets.utils import (
+    _adjust_df_date,
+    _get_dataset_path,
+    _parse_interval,
+    _validate_timestamp,
+)
+
+logger = getLogger(__name__)
+
+
+base_config = BaseConfig()
+
+
+@dataclass(init=False)
+class Weather(Dataset):
+    """Weather Forecast Dataset
+
+    The Weather Forecast Dataset contains meteorological features at a particular place (defined by latitude and longitude features) and time.
+    This dataset can present data distribution shifts over both time and space.
+
+    The original data was sourced from the `Weather Prediction Dataset <https://github.com/Shifts-Project/shifts>`_.
+    From the source data additional transformations were made, such as: feature renaming, feature selection and subsampling.
+    The original dataset is described in
+    `Shifts: A Dataset of Real Distributional Shift Across Multiple Large-Scale Tasks <https://arxiv.org/pdf/2107.07455.pdf>`_, by **Malinin, Andrey, et al.**
+
+    For a detailed description, please use the dataset's describe() method or visit whylog's documentation website.
+    """
+
+    baseline_df: pd.DataFrame
+    inference_df: pd.DataFrame
+    inference_interval: str = "1d"
+    number_days: int = 1
+    unit: str = "D"
+    url: str = WeatherConfig.url
+    baseline_timestamp: Union[date, datetime] = date.today()
```

Comment:
Do we need timezone info here?

Reply:
Yes!
I'll have to do some refactoring, because whylogs expects datetime objects, and the default for this dataset is date, which is naive. I'll just clear hour,seconds, etc to represent dates as datetimes and set to UTC timezone so it is compatible with whylogs' other modules.

Reply:
Done in the latest commit.
everything is in datetime format now. If the user doesn't pass any timezone info, it is assumed local and then converted to UTC timezone. In the dataframe, dates always contain time zone info.